### PR TITLE
Adapt to Atmosphere 0.10.0 and libnx 3.0.0

### PR DESCRIPTION
**Pull request by cathery**
- Changed all the namings in accordance with libnx 3.0.0
- Changed the file locations of Atmosphere's config and sysmodules
- Kosmos Toolbox now creates and fills the config file if none was found.
- Allow inputs from any controller, for consistency with hbmenu v3.2.0

### CODE REVIEW

File: source/threads.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@ Result Threads::create(ThreadFunc func) {
   Thread thread;
 
   Result res;
-  if (R_FAILED( res = threadCreate(&thread, func, nullptr, 0x2000, 0x2B, -2)))
+  if (R_FAILED( res = threadCreate(&thread, func, nullptr, NULL, 0x2000, 0x2B, -2)))
```
Comment (WerWolv) [inline source/threads.cpp]: Should be nullptr instead of NULL like the parameter before

File: include/ini/simpleIniParser/ini.hpp [hunk 2]
```diff
@@ -30,7 +30,9 @@ namespace simpleIniParser {
             ~Ini();
             std::string build();
             IniSection * findSection(std::string name);
+            IniSection * findOrCreateSection(IniSectionType type, std::string name);
```
Comment (WerWolv) [inline include/ini/simpleIniParser/ini.hpp]: These functions should probably be added to https://github.com/AtlasNX/SimpleIniParser as well. Tbh I'm not sure why I didn't just just made a submodule out of this

File: Makefile [hunk 3]
```diff
@@ -38,12 +38,12 @@ APP_TITLE	:=	Kosmos Toolbox
 APP_AUTHOR	:=	WerWolv
 APP_VERSION	:=	${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_MICRO}
 
-TARGET		:=	$(notdir $(CURDIR))
-OUTDIR    	:= out
+TARGET		:=	KosmosToolbox
+OUTDIR		:=	out
 BUILD		:=	build
-SOURCES		:=	source source/ini
+SOURCES		:=	source source/ini source/GUI source/Utils
```
Comment (WerWolv) [inline Makefile]: I usually name my folders in snake_case so imo `gui` and `utils` would be better

**Review by WerWolv — CHANGES_REQUESTED** [review]

**Comment by WerWolv** [comment]
Also a long standing bug that I never got around to fix is that the toolbox doesn't create the `/atmosphere/contents/<titleID>/flags` folder before attempting to create the boot2.flag file so if the folder isn't there, the flag just won't be created. If you have time I'd be really grateful if you could fix that while you're on it.
Thanks a lot!

File: source/threads.cpp (hunk 1 above)
Comment (cathery) [reply source/threads.cpp]:
```suggestion
  if (R_FAILED( res = threadCreate(&thread, func, nullptr, nullptr, 0x2000, 0x2B, -2)))
```

**Review by cathery — COMMENTED** [review]
pam pam

File: source/Utils/threads.cpp [hunk 4]
```diff
@@ -8,7 +8,7 @@ Result Threads::create(ThreadFunc func) {
   Thread thread;
 
   Result res;
-  if (R_FAILED( res = threadCreate(&thread, func, nullptr, 0x2000, 0x2B, -2)))
+  if (R_FAILED( res = threadCreate(&thread, func, nullptr, NULL, 0x2000, 0x2B, -2)))
```
Comment (cathery) [inline source/Utils/threads.cpp]:
```suggestion
  if (R_FAILED( res = threadCreate(&thread, func, nullptr, nullptr, 0x2000, 0x2B, -2)))
```

File: source/GUI/gui_sysmodule.cpp [hunk 5]
```diff
@@ -114,12 +114,12 @@ GuiSysmodule::GuiSysmodule() : Gui() {
         u64 tid = std::stol(sysmodule.first.c_str(), nullptr, 16);
```
Comment (WerWolv) [inline source/GUI/gui_sysmodule.cpp]:
```suggestion
        mkdir(std::string("/atmosphere/contents/" + sysmodule.second.titleID + "/flags").c_str(), 777);
```

File: include/ini/simpleIniParser/ini.hpp (hunk 2 above)
Comment (cathery) [reply include/ini/simpleIniParser/ini.hpp]: Waiting for [PR from SimpleIniParser](https://github.com/AtlasNX/SimpleIniParser/pull/2) now

**Review by cathery — COMMENTED** [review]
a

**Pull request merged**